self_testim.py

```diff
@@ -0,0 +1,41 @@
+def is_valid(board, neighbor, initial_point):  # neighbor[0]=row index, neighbor[1]=col index,
+    return 0 <= neighbor[0] < len(board) and 0 <= neighbor[1] < len(board[neighbor[0]]) \
+           and board[neighbor[0]][neighbor[1]] != -1 and [neighbor[0], neighbor[1]] > initial_point
```

Comment:
Invalid code `[neighbor[0], neighbor[1]] > initial_point`

Reply:
You never set something to `-1`, this condition has no meaning that way.
In general it is not wise to use numbers as special cases, put `None` for special case